app/liquid_tags/runkit_tag.rb

```diff
@@ -1,15 +1,16 @@
 class RunkitTag < Liquid::Block
   def initialize(tag_name, markup, tokens)
     super
-    @markup = markup
+    @preamble = ActionView::Base.full_sanitizer.sanitize(markup, tags: [])
   end
 
   def render(context)
     content = Nokogiri::HTML.parse(super)
     parsed_content = content.xpath("//html/body").text
     html = <<~HTML
       <div class="runkit-element">
-        #{parsed_content}
+        <code style="display: none">#{@preamble}</code>
```

Comment:
Note I added an inline `display: none` here on the basis that if a user didn't have JS running, they still should not see the preamble code, as the author didn't intend for it to be visible.